src/main/kotlin/com/jetbrains/snakecharm/inspections/quickfix/IntroduceArgument.kt

```diff
@@ -0,0 +1,50 @@
+package com.jetbrains.snakecharm.inspections.quickfix
+
+import com.intellij.codeInsight.CodeInsightUtilCore
+import com.intellij.codeInsight.template.TemplateBuilderFactory
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.jetbrains.python.psi.LanguageLevel
+import com.jetbrains.python.psi.PyElementGenerator
+import com.jetbrains.python.psi.PyKeywordArgument
+import com.jetbrains.python.psi.impl.PyPsiUtils
+import com.jetbrains.snakecharm.SnakemakeBundle
+
+class IntroduceArgument(element: PsiElement) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
+    private val defaultArgumentName = "arg"
+
+    override fun getFamilyName() = SnakemakeBundle.message("INSP.INTN.name.argument")
+
+    override fun getText() = SnakemakeBundle.message("INSP.INTN.name.argument")
+
+    override fun invoke(
+            project: Project,
+            file: PsiFile,
+            editor: Editor?,
+            startElement: PsiElement,
+            endElement: PsiElement
+    ) {
+        var element = startElement
+        PyPsiUtils.assertValid(element)
+        if (element.isValid) {
+            val elementGenerator = PyElementGenerator.getInstance(project)
+            val argument = elementGenerator.createKeywordArgument(
+                    LanguageLevel.forElement(element),
+                    defaultArgumentName,
+                    element.text
+            )
+
+            element = element.replace(argument)
+            if (element == null) return
+            element = CodeInsightUtilCore.forcePsiPostprocessAndRestoreElement(element)
+            if (element == null) return
+            val builder = TemplateBuilderFactory.getInstance().createTemplateBuilder(element)
+            val keywordArgument = (element as PyKeywordArgument).keywordNode?.psi!!
```

Comment:
Could class cast exception occur here?  `IntroduceArgument()` signature is generic and allow user to use it  for any Psi.

Reply:
At this point line `element = element.replace(argument)` has already been executed, and `argument` is created as `PyKeywordArgument`. This `element` is not the original element, it's a keyword argument named `arg` that has the text of the original element for a value. So, if I understand correctly, there will be no class exception; however, if by this comment you mean I should probably make signature more restricting, then I'll change it to `PyExpression`.

Reply:
And if a user enters some invalid text in red box (instead of identification), element will be still PyKeywordArgument?

Reply:
It won't, but it's not the element in question. The element that gets cast to `PyKeywordArgument` is not the one user performs modifications on (yet). We created a keyword argument from 'arg = somePyExpression'. We want to get a PSI element for `arg` from this freshly created keyword argument. We get this PSI element and we tell `builder` which element should be put in that red box in which the user can enter a new name of their choice. At the moment where the cast is executed, no red box or user input exist yet, only a PSI element which we have created from text and which must be a `PyKeywordArgument` object.